src/lib/core/TLVReader.cpp

```diff
@@ -345,7 +345,7 @@ CHIP_ERROR TLVReader::Get(CharSpan & v) const
     }
 
     v = CharSpan(Uint8::to_const_char(bytes), len);
-#if CHIP_CONFIG_TLV_VALIDATE_CHAR_STRING_ON_READ
+#if defined(CHIP_CONFIG_TLV_VALIDATE_CHAR_STRING_ON_READ) && CHIP_CONFIG_TLV_VALIDATE_CHAR_STRING_ON_READ
```

Comment:
This one is defined by lib/core/CHIPConfig.h (actually buildconfig in gn).

We should never have it undefined. Not having this means that the chip config is old or not properly defined.